### ISSUE TITLE
Feature request - Add children to `Add` component - Or another way to create a component at the same position

### ISSUE THREAD

**Issue by Vadorequest** [reaviz/reaflow#46]
I could really use a menu when clicking on a connection `Add` component.

![image](https://user-images.githubusercontent.com/3807458/107072390-56b87400-67e6-11eb-8a75-1cca274bd886.png)

When clicking on "+", display a custom component.

Here's what I've done for now instead:
![image](https://user-images.githubusercontent.com/3807458/107071557-29b79180-67e5-11eb-93ae-6f02dc2e97bf.png)

Right now, I'm displaying a list of components at the bottom to pick from, but it's not so easy:
- I decided to use an absolute position, but ideally, I'd prefer to display this at the spot it was clicked. But, I don't know what are the x/y of the "Add" (they're not exposed). Even though they were exposed, I'd have a hard time converting it to an absolute css position (top/left/bottom/right).
- Using an absolute position is easier. I thought about using another element within the canvas, but that'd have to be a node, right? And this doesn't really make sense, it'd be some kind of temporary node 🤔 and it'd move all other nodes/edges upon display. So, using something displayed in absolute position seemed better.

Eventually, what I'm trying to achieve is something of the sort: https://youtu.be/frihIydsxAg
Basically, click on the `Add` element, display a menu, click on an element and get the new node added.

I feel like I can't do this at this time because:
- `Add` doesn't allow defining `children`, this might be the easiest way to position an element in the Canvas at the right position, that displays on top of other elements and wouldn't need to redraw the whole canvas.
- Expose x/y in the `onAdd` of the `Edge` component
  - Current signature is `onAdd?: (event: React.MouseEvent<SVGGElement, MouseEvent>, edge: EdgeData) => void;`, might become something like `onAdd?: (event: React.MouseEvent<SVGGElement, MouseEvent>, edge: EdgeData, meta: { x:number; y:number }) => void;`	

I don't think I can easily achieve the above at this time. 
Would it make sense to add native support for this?

**Comment by Vadorequest** [comment]
I managed to achieve what I wanted, but I had to do it very differently. (not only because of Reaflow's limitations, but also because I wanted to achieve a particular UX).

I disabled the `Remove` built-in component and customised the `Add` component because I wanted both Add/Remove buttons to display. By default, when using both of them they'll appear at the same X/Y position and will be on top of each other, displaying only one of them. I wanted both.

I figured what I really want to do is to have a custom `children` in the Edge component, to display a custom component when the Edge is being selected. This is currently not possible and tracked through https://github.com/reaviz/reaflow/issues/67.

> I'll close this issue in favor of https://github.com/reaviz/reaflow/issues/67.

---

Result:
![image](https://user-images.githubusercontent.com/3807458/109287436-d785f100-7823-11eb-8180-7d970f4662c3.png)

```
import React, { Fragment } from 'react';
import {
  Add,
  AddProps,
  Remove,
  RemoveProps,
} from 'reaflow';
import BaseEdgeData from '../../types/BaseEdgeData';

type Props = {
  hidden: boolean;
  onRemove?: (
    event: React.MouseEvent<SVGGElement, MouseEvent>,
    edge: BaseEdgeData,
  ) => void;
  onAdd?: (
    event: React.MouseEvent<SVGGElement, MouseEvent>,
    edge: BaseEdgeData,
  ) => void;
} & Partial<AddProps | RemoveProps>;

/**
 * Displays the edge's actions (add node, remove current edge).
 *
 * Displays the actions only when the edge is being selected.
 * Displays the actions a bit above the edge, in a line.
 */
const EdgeActions: React.FunctionComponent<Props> = (props) => {
  const {
    x,
    y,
    onAdd,
    onRemove,
  } = props;
  const aboveLineY = (y || 0) - 20;
  const removeX = (x || 0) + 20;

  // console.log('EdgeActions props', props);

  return (
    <Fragment>
      <Add
        {...props}
        y={aboveLineY}
        // @ts-ignore
        onClick={onAdd}
      />
      <Remove
        {...props}
        x={removeX}
        y={aboveLineY}
        // @ts-ignore
        onClick={onRemove}
      />
    </Fragment>
  );
};

export default EdgeActions;

```

Custom Edge component:
```tsx
    <Edge
      {...props}
      className={classnames(`edge`, { 'is-selected': isSelected })}
      add={(
        <EdgeActions
          hidden={!isSelected}
          onAdd={onAddIconClick}
          onRemove={onRemoveIconClick}
        />
      )}
      onClick={onEdgeClick}
    />
```

Code:
- https://github.com/Vadorequest/poc-nextjs-reaflow/blob/5e842bd1145a0a7b0c8c73cba9a36d81a9aee791/src/components/edges/EdgeActions.tsx#L35
- https://github.com/Vadorequest/poc-nextjs-reaflow/blob/7524892d92e585c9cf9f240b7da1e1861d1c22e1/src/components/edges/BaseEdge.tsx#L155
